module/enrichers.mjs

```diff
@@ -102,6 +108,150 @@ function enrichDND5ESkill([match, terms]) {
   return tag;
 }
 
+/* -------------------------------------------- */
+/*  Awards                                      */
+/* -------------------------------------------- */
+
+/**
+ * Parse an award's terms into an object representing what should be gained
+ * @param {string} terms
+ * @returns {Record<string, number>}
+ */
+function parseAwardTerms(terms) {
+  const pattern = new RegExp(/^(.+?)(\D+)$/);
+  const currency = {};
+  const invalid = [];
+  let milestones;
+  for ( const part of terms.split(" ") ) {
+    if ( !part ) continue;
+    let [, amount, label] = part.match(pattern) ?? [];
+    label = label?.toLowerCase();
+    try {
+      new Roll(amount);
+      if ( label in crucible.CONFIG.currency ) currency[label] = amount;
+      else if ( ["xp", "milestone", "milestones"].includes(label) ) milestones = Number(amount);
+      else throw new Error();
+    } catch (err) {
+      invalid.push(part);
+    }
+  }
+
+  if ( invalid.length ) throw new Error(game.i18n.format("AWARD.InvalidWarning", {
+    terms: game.i18n.getListFormatter().format(invalid.map(i => `"${i}"`))
+  }));
+
+  return { currency, milestones }
+}
+
+/**
+ * Enrich an Award with the format [[/award {...awards}]]
+ * @param {string} match
+ * @param {string} terms
+ */
+function enrichAward([match, terms]) {
+  const { currency, milestones } = parseAwardTerms(terms);
+  const entries = [];
+  for ( let [currencyKey, amount] of Object.entries(currency) ) {
+    const {label, icon, abbreviation} = crucible.CONFIG.currency[currencyKey];
+    const textOrIcon = icon ? `<i class="currency ${currencyKey}" data-tooltip aria-label="${label}" style="background-image: url(${icon});"></i>` : abbreviation;
+    entries.push(`
+      <span class="award-entry">
+        ${amount} ${textOrIcon}
+      </span>
+    `);
+  }
+  if ( milestones ) {
+    const plurals = new Intl.PluralRules(game.i18n.lang);
+
+    entries.push(`
+      <span class="award-entry">
+        ${milestones} ${game.i18n.localize("AWARD.Milestone." + plurals.select(milestones))}
+      </span>
+  `);
+  }
+  // Return the enriched content tag
+  const tag = document.createElement("enriched-content");
+  tag.classList.add("award");
+  tag.dataset.terms = terms;
```

Comment:
We should add more direct data here so we don't have to re-parse the string on click.